x-pack/filebeat/input/awss3/sqs_test.go

```diff
@@ -284,3 +284,36 @@ func TestSQSReaderLoop(t *testing.T) {
 func TestSQSWorkerLoop(t *testing.T) {
 
 }
+
+func TestCancelWithGrace(t *testing.T) {
+	// TODO: Rewrite this to use testing/synctest when it is available without
+	// GOEXPERIMENT=synctest. See https://go.dev/blog/synctest.
+
+	const (
+		wait    = time.Second
+		tooLong = time.Second
+		tol     = 10 * time.Millisecond
+	)
+	parentCtx, parentCancel := context.WithCancel(context.Background())
+	childCtx, childCancel := cancelWithGrace(parentCtx, wait)
+	defer childCancel()
+
+	var parentCancelled, childCancelled time.Time
+	parentCancel()
+	select {
+	case <-time.After(tooLong):
+		t.Fatal("parent context failed to cancel within timeout")
+	case <-parentCtx.Done():
+		parentCancelled = time.Now()
```

Comment:
[Suggestion]
`parent` is already cancelled, you don't need the wait here. If `<-parentCtx.Done()` does not get selected, you could fail the test immediately.

Reply:
Yeah, it's here for symmetry.